internal/fnruntime/wasm.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fnruntime
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+type WasmRuntime string
+
+const (
+	Wasmtime WasmRuntime = "wasmtime"
+	Nodejs   WasmRuntime = "nodejs"
+
+	WasmRuntimeEnv = "KPT_FN_WASM_RUNTIME"
+
+	jsEntrypointFunction = "processResourceList"
+)
+
+type WasmFn struct {
+	runtimeType WasmRuntime
+
+	wasmtime *WasmtimeFn
+	nodejs   *WasmNodejsFn
+}
+
+func NewWasmFn(wasm string) (*WasmFn, error) {
+	switch os.Getenv(WasmRuntimeEnv) {
+	case string(Nodejs):
+		nf, err := NewNodejsFn(wasm)
+		if err != nil {
+			return nil, err
+		}
+		return &WasmFn{
+			runtimeType: Nodejs,
+			nodejs:      nf,
+		}, nil
+	case string(Wasmtime):
+		fallthrough
```

Comment:
I think it would be fine to default to "wasmtime" if the env var is not set at all. But if it is set to something besides "wasmtime" or "nodejs" (like "foo") we should probably throw an error.

Reply:
Done